src/shared/logger/activation.ts

```diff
@@ -198,3 +202,62 @@ async function registerLoggerCommands(context: vscode.ExtensionContext): Promise
         )
     )
 }
+
+/**
+ * Deletes the older logs when there are too many or are using too much space.
+ */
+async function cleanLogFiles(): Promise<void> {
+    let files = await fs.readdir(logPath)
+
+    if (files.length > 100) {
+        deleteOldLogFiles(files, 10)
+        files = await fs.readdir(logPath)
+    }
+
+    let dirSize = 0
+    const oversizedFiles = []
+    for (const log of files) {
+        try {
+            const logSize = (await fs.stat(path.join(logPath, log))).size
```

Comment:
- the try-catch should only surround the fs.stat() call
- store the join() result in a `const logFullPath`